src/main/java/org/graylog/integrations/notifications/types/SlackEventNotification.java

```diff
@@ -0,0 +1,236 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Graylog.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.graylog.integrations.notifications.types;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.floreysoft.jmte.Engine;
+import org.apache.directory.api.util.Strings;
+import org.graylog.events.notifications.*;
+import org.graylog.events.processor.EventDefinitionDto;
+import org.graylog.events.processor.aggregation.AggregationEventProcessorConfig;
+import org.graylog.integrations.notifications.modeldata.StreamModelData;
+import org.graylog.scheduler.JobTriggerDto;
+import org.graylog2.jackson.TypeReferences;
+import org.graylog2.notifications.Notification;
+import org.graylog2.notifications.NotificationService;
+import org.graylog2.plugin.MessageSummary;
+import org.graylog2.plugin.streams.Stream;
+import org.graylog2.plugin.system.NodeId;
+import org.graylog2.shared.bindings.providers.OkHttpClientProvider;
+import org.graylog2.streams.StreamService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+import java.util.*;
+import java.util.stream.Collectors;
+
+import static com.google.common.base.Strings.isNullOrEmpty;
+import static java.util.Objects.isNull;
+import static java.util.Objects.requireNonNull;
+
+public class SlackEventNotification implements EventNotification {
+
+	private static final String UNKNOWN_VALUE = "<unknown>";
+
+	public interface Factory extends EventNotification.Factory {
+		@Override
+        SlackEventNotification create();
+	}
+
+
+	private static final Logger LOG = LoggerFactory.getLogger(SlackEventNotification.class);
+
+	private final EventNotificationService notificationCallbackService;
+	private final Optional<StreamService> streamService ;
+	private final Engine templateEngine ;
+	private final NotificationService notificationService ;
+	private final ObjectMapper objectMapper ;
+	private final NodeId nodeId ;
+	private final OkHttpClientProvider okHttpClientProvider ;
+
+
+	@Inject
+	public SlackEventNotification(EventNotificationService notificationCallbackService,
+								  ObjectMapper objectMapper,
+								  Engine templateEngine,
+								  NotificationService notificationService,
+								  OkHttpClientProvider okHttpClientProvider,
+								  NodeId nodeId, StreamService streamService){
+		this.notificationCallbackService = notificationCallbackService;
+		this.objectMapper = requireNonNull(objectMapper);
+		this.templateEngine = requireNonNull(templateEngine);
+		this.okHttpClientProvider = requireNonNull(okHttpClientProvider);
+		this.notificationService = requireNonNull(notificationService);
+		this.nodeId = requireNonNull(nodeId);
+		this.streamService = Optional.ofNullable(streamService);
```

Comment:
we do not want to deal with null checks. also, the streams is `optional` in the event definitions. I think, it should be `Optional`.

Reply:
See above comment.  `Optional` was added to Java to make it easier to handle instances where `null` may be returned instead of an object.  Guice is not going to inject a `null` value for StreamService.  Therefore there is no reason to use `Optional`.